### DISCUSSION TITLE
Draw dotted lines instead of gaps for missing values

### DISCUSSION THREAD

**Discussion by crazysurajit** [highfestiva/finplot#287] [Q&A]
Could you please advise how to draw dotted lines instead of gaps using finplot?

For matplotlib this solution works: https://stackoverflow.com/questions/45176584/dotted-lines-instead-of-a-missing-value-in-matplotlib

**Comment by highfestiva** [comment]
See examples.

**Comment by crazysurajit** [comment]
Thank you, finally got it working using `add_line` in a `for` loop.

However was expecting, as this worked in matplotlib
```
import numpy as np
import matplotlib.pyplot as plt
y = np.array([np.NAN, 45, 23, np.NAN, 5, 14, 22, np.NAN, np.NAN, 18, 23])
x = np.arange(0, len(y))
mask = np.isfinite(y)

fig, ax = plt.subplots()
(line,) = ax.plot(x[mask], y[mask], ls="--", lw=1)
ax.plot(x, y, color=line.get_color(), lw=1.5)
plt.show()
```

this would work too in finplot
```
import numpy as np
import finplot as fplt
y = np.array([np.NAN, 45, 23, np.NAN, 5, 14, 22, np.NAN, np.NAN, 18, 23])
x = np.arange(0, len(y))
mask = np.isfinite(y)

fplt.create_plot()
fplt.plot(x[mask], y[mask], style="--")
fplt.plot(x, y)
fplt.show()
```

**Reply by highfestiva** [reply]
If you're not using a complete index, you need to plot the curve that has the complete index first (as a frame of reference):
```python
fplt.plot(x, y)
fplt.plot(x[mask], y[mask], style="--")
```
This would however only plot one line on top of the other. Probably not what you're after.
